### PR TITLE
Issue #10157 Resolve aliases in getAlias() instead of setAlias()

### PR DESCRIPTION
**Pull request by mikehaertl**
| Q | A |
| --- | --- |
| Is bugfix? | no |
| New feature? | yes |
| Breaks BC? | yes |
| Tests pass? | yes/no |
| Fixed issues | #10157 |

### CODE REVIEW

**Comment by Faryshta** [comment]
can you add a unit test for the scenario i described on #10157 ?

**Comment by mikehaertl** [comment]
Added some unit tests. It should cover most cases.

File: framework/BaseYii.php [hunk 1]
```diff
@@ -138,11 +138,13 @@ public static function getAlias($alias, $throwException = true)
 
         if (isset(static::$aliases[$root])) {
             if (is_string(static::$aliases[$root])) {
-                return $pos === false ? static::$aliases[$root] : static::$aliases[$root] . substr($alias, $pos);
+                $path = $pos === false ? static::$aliases[$root] : static::$aliases[$root] . substr($alias, $pos);
+                return strncmp($path, '@', 1) ? $path : static::getAlias($path);
```
Comment (samdark) [inline framework/BaseYii.php]: Result of `strncmp` should be compared with `=== 0`.

**Comment by Sxderp** [comment]
Should the Alias be re-fetched every time a call is made to getAlias? I don't think it should be, I think it adds unnecessary complexity. I can't think of a valid use case for wanting this functionality. Actually, I can't really think of a use case for changing an alias while the application is running (that is, after the configs have been loaded).

File: framework/BaseYii.php (hunk 1 above)
Comment (mikehaertl) [inline framework/BaseYii.php]: It's already used without `===0` on other places. That's why I've used the same pattern. If we change this we should change it everywhere.

**Comment by samdark** [comment]
Won't make it into 2.0. There's separate alias component in 3.0 now.

**Pull request closed without merging**